expression/builtin_cast_vec.go

```diff
@@ -293,10 +293,13 @@ func (b *builtinCastTimeAsDecimalSig) vecEvalDecimal(input *chunk.Chunk, result
 		if result.IsNull(i) {
 			continue
 		}
-		times[i].FillNumber(&decs[i])
-		if _, err := types.ProduceDecWithSpecifiedTp(&decs[i], b.tp, sc); err != nil {
+		dec := new(types.MyDecimal)
```

Comment:
Please define this variable outside this loop so that we don't need to create a new object each time.

Reply:
> Please define this variable outside this loop so that we don't need to create a new object each time.

It's done. Please take a look again.